ProtonBeamDumpGeom/inc/ProtonBeamDump.hh

```diff
@@ -23,52 +23,60 @@ namespace mu2e {
   public:
 
     //----------------------------------------------------------------
-    // Primary inputs: sizes
+    // Beam dump core placement and dimensions
 
-    const std::vector<double>& coreHalfSize() const { return _coreHalfSize; }
-    const std::vector<double>& neutronCaveHalfSize() const { return _neutronCaveHalfSize; }
-    const std::vector<double>& mouthHalfSize() const { return _mouthHalfSize; }
-    const std::vector<double>& frontSteelHalfSize() const { return _frontSteelHalfSize; }
-    const std::vector<double>& backSteelHalfSize() const { return _backSteelHalfSize; }
-    const std::vector<double>& coreAirHalfSize() const { return _coreAirHalfSize; }
-
-    double minCoreShieldingThickness() const { return _minCoreShieldingThickness; }
-
-    //----
-    // Primary inputs: placement
     const CLHEP::Hep3Vector& coreCenterInMu2e() const { return _coreCenterInMu2e; }
     // absolute w.r.t to the Mu2e
     double coreRotY() const { return _coreRotY; }
     const CLHEP::HepRotation& coreRotationInMu2e() const { return _coreRotationInMu2e; }
 
-    //---
-    // Derived stuff
-
-    // Front core shielding: contains the dump core and collimator1.
-    double coreCenterDistanceToShieldingFace() const { return _coreCenterDistanceToShieldingFace; }
-    double coreCenterDistanceToReferencePlane() const { return _coreCenterDistanceToReferencePlane; }
+    const std::vector<double>& coreHalfSize() const { return _coreHalfSize; }
 
-    const std::vector<double>& frontShieldingHalfSize() const { return _frontShieldingHalfSize; }
-    const CLHEP::Hep3Vector& frontShieldingCenterInMu2e() const { return _frontShieldingCenterInMu2e; }
+    //----------------------------------------------------------------
+    // The beam dump concrete poured around the core.  It is modeled
+    // as a boolean subtraction volume.  The starting volume is as an
+    // extruded solid parameterized here.  Some subtraction pieces are
+    // defined below: the "mouth" and the "neutron cave", as well as
+    // the ExtMon room cutout.  Another subtraction is based on the
+    // Entrace collimator for the Extinction Monitor, its parameters
+    // are not provided by this class.
+    //
+    // The parameters for the mouth and neutron cave subtraction
+    // volumes here are nominal.  The margins to ensure proper
+    // overlaps for G4 booleans are added by the code talking to G4.
+    // OTOH extMonSubtractionOutline is tweaked here, as postponing
+    // the extruded solid tweak would make the overall code too entangled.
+
```

Comment:
Technically, you don't need a const accessor if you return by value in this function. Is there a reason you return by value and not reference?

Reply:
Good catch.  I will look at the header and clean it up, changing to const ref returns.